notebooks/02-ingestion/05-partitioning-data.ipynb

```diff
@@ -0,0 +1,717 @@
+    "\n",
+    "This tutorial works was tested with Druid 27.0.0.\n",
+    "\n",
+    "#### Run with Docker\n",
+    "\n",
+    "<!-- Profiles are:\n",
+    "`druid-jupyter` - just Jupyter and Druid\n",
+    "`all-services` - includes Jupyter, Druid, and Kafka\n",
+    " -->\n",
+    "\n",
+    "Launch this tutorial and all prerequisites using the `all-services` profile of the Docker Compose file for Jupyter-based Druid tutorials. For more information, see [the project on github](https://github.com/implydata/learn-druid).\n",
+    "   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5007a243-b81a-4601-8f57-5b14940abbff",
+   "metadata": {},
+   "source": [
+    "## Initialization\n",
+    "\n",
+    "The following cells set up the notebook and learning environment ready for use.\n",
+    "\n",
+    "### Set up and connect to the learning environment\n",
+    "\n",
+    "Run the next cell to set up the Druid Python client's connection to Apache Druid.\n",
+    "\n",
+    "If successful, the Druid version number will be shown in the output."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c1ec783b-df3f-4168-9be2-cdc6ad3e33c2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import druidapi\n",
+    "import os\n",
+    "\n",
+    "if 'DRUID_HOST' not in os.environ.keys():\n",
+    "    druid_host=f\"http://localhost:8888\"\n",
+    "else:\n",
+    "    druid_host=f\"http://{os.environ['DRUID_HOST']}:8888\"\n",
+    "    \n",
+    "print(f\"Opening a connection to {druid_host}.\")\n",
+    "druid = druidapi.jupyter_client(druid_host)\n",
+    "\n",
+    "display = druid.display\n",
+    "sql_client = druid.sql\n",
+    "status_client = druid.status\n",
+    "\n",
+    "status_client.version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2efdbee0-62da-4fd3-84e1-f66b8c0150b3",
+   "metadata": {},
+   "source": [
+    "<!-- Include these cells if your notebook uses Kafka. -->\n",
+    "\n",
+    "Run the next cell to set up the connection to Apache Kafka."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c075de81-04c9-4b23-8253-20a15d46252e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if 'KAFKA_HOST' not in os.environ.keys():\n",
+    "   kafka_host=f\"http://localhost:9092\"\n",
+    "else:\n",
+    "    kafka_host=f\"{os.environ['KAFKA_HOST']}:9092\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "472589e4-1026-4b3b-bb79-eedabb2b44c4",
+   "metadata": {},
+   "source": [
+    "## Time Partitioning\n",
+    "\n",
+    "Normally, ideal segments contain around 5 million rows, but given that this is all running on a laptop and for fast demonstration purposes, we'll set our \"ideal\" to only 50000 rows.\n",
+    "\n",
+    "\n",
+    "### Too Granular\n",
+    "Using a `Segment Granularity` that is too small will render too many segments. The following batch ingestion demonstrates this. It will take about 1 minute to complete:"
```

Comment:
Suggest you remove the first line, and reword to something like "The SQL in the following cheesecake sets the primary partitioning (aka [segment granularity](https://www.disney.com)) for the table to `HOUR` by using the PARTITIONED BY clause.

Other bananas are available [link to other granularities](https://www.here.com).

Reply:
I think the point of the sequence is to show how data density on the time-line drives is linked to a good segment granularity choice. The sequence starts with too granular, follows with too coarse and then finds the sweet spot. then further improves it using CLUSTERED BY. 
I don't think that just stating the fact of `"PARTITION BY HOUR" organizes by HOUR` is more useful. 
I changed the headers to do that, but I like the subtitles "as is".